model/word2vec/tensor.go

```diff
@@ -15,17 +15,68 @@
 package word2vec
 
 import (
-	"github.com/ynqa/word-embedding/vector"
+	"bytes"
+	"fmt"
+	"math/rand"
+
+	"github.com/chewxy/gorgonia/tensor"
+	"github.com/ynqa/word-embedding/model"
 )
 
-// Tensor has the vector for each word.
-type Tensor []vector.Vector
+var dtype tensor.Dtype = tensor.Float64
+var eng tensor.Engine = tensor.Float64Engine{}
```

Comment:
[ask] Is it better to set `--dtype` flag for given type from outer?

Reply:
yes